### PR TITLE
[MAINTENANCE] Use `Comparable` type alias for Expectation min/max value

### PR DESCRIPTION
**Pull request by cdkini**
Alias for comparable to DRY up code and added `date` as a valid input

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/core/types.py
Comment (Kilo59) [inline great_expectations/core/types.py:1]: What happened to the old `aliases` module we had for stuff like this?

Comment (cdkini) [reply great_expectations/core/types.py:1]: Oh I have no idea 🫨

**Review by tyler-hoffman — APPROVED** [review]
Looks like this alters a number of expectations, adding the option of date wen it previously only allowed date-time. I think it's fine, but just noting.

**Comment by cdkini** [comment]
@tyler-hoffman yup that was part of the AC of the ticket - think we wanted to expand the comparable

**Review by Kilo59 — CHANGES_REQUESTED** [review]
This should go here
https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/alias_types.py

File: great_expectations/core/types.py
Comment (Kilo59) [reply great_expectations/core/types.py:1]: https://github.com/great-expectations/great_expectations/blob/develop/great_expectations/alias_types.py

**Pull request merged**